### PR TITLE
WiP: Towards ScientificData-compliant output

### PR DESCRIPTION
**Pull request by mih**
This PR aims to eventually address all outstanding issues (see those listed below for a summary of what needs to be accomplished). As most of them are heavily inter-dependent this will be approached in one go. No intermediate commit should break existing functionality.
- [x] #3
- [x] #4
- [x] #5
- [x] #6
- [x] #7
- [x] #8
- [x] #9
- [x] #10
- [x] #11
- [x] #12
- [x] #13 

Please don't merge while still titled "WiP: ...". Please comment on this at any point.

### CODE REVIEW

**Comment by mih** [comment]
All initially identified issues have been addressed -- now waiting for review.

File: bids2isatab/main.py [hunk 1]
```diff
@@ -8,61 +8,177 @@
...
+sample_property_name_map = {
+    "age": "Characteristics[age at scan]",
+    "gender": "Characteristics[sex]",
+    "handedness": "Characteristics[handedness]",
+    "participant_id": "Sample Name",
+    "sex": "Characteristics[sex]",
+}
+
+
+def get_bids_metadata(bids_root, basepath):
```
Comment (chrisgorgo) [inline bids2isatab/main.py:121]: This could be replaced by a pybids.grabbids call.

File: bids2isatab/main.py [hunk 2]
```diff
@@ -81,124 +198,432 @@ def get_keychains(d, dest, prefix):
...
-        other_fields.append(get_metadata_for_nifti(args.bids_directory, file))
-
-    assay_dict["Sample Name"] = sample_names
-    assay_dict["Protocol REF"] = "Magnetic Resonance Imaging"
-    assay_dict["Parameter Value[Modality]"] = types
-    mri_par_names.append("Modality")
-    assay_dict["Parameter Value[Resolution]"] = resolutions
-    mri_par_names.append("Resolution")
-    assay_dict["Unit"] = resolutions_units
+def _get_file_matches(bids_directory, glob_pattern):
```
Comment (chrisgorgo) [inline bids2isatab/main.py:320]: this also potentially could be replaced by pybids.grabbids call

**Review by chrisgorgo — APPROVED** [review]
Other than potential use of pybids this looks good to me.

**Pull request merged**